File: src/server/BuildingBlock/MovieTicket.Domain/Entities/Reservation.cs
using MovieTicket.Core.Domain;

namespace MovieTicket.Domain.Entities;

public class Reservation : EntityBase
{
    public Guid ScreeningId { get; set; }
    public Guid UserId { get; set; }
    
    public long TotalPrice { get; set; }
    
    public long ItemPrice { get; set; }

    public string TransactionId { get; set; }
    
    public ReservationState ReservationState { get; set; }
    
    public virtual Screening Screening { get; set; }
    public virtual ICollection<SeatReservation> SeatReservations { get; set; }
    public virtual ICollection<ServiceReservation> ServiceReservations { get; set; }
    

}

File: src/client/mobile/app/src/main/java/com/superman/movieticket/domain/entities/Reservation.kt
package com.superman.movieticket.domain.entities

import com.superman.movieticket.core.entity.BaseEntity
import java.io.Serializable
import java.util.Date

data class Reservation (
    var id :String,
    var screeningId: String,
    var totalPrice: Long,
    var itemPrice: Long,
    var screening: Screening,
    var seatReservations: MutableList<SeatReservation>,
    var serviceReservations: MutableList<ServiceReservation>,
    var createdDate : String= Date().toString()
) : Serializable
